packages/react-router/src/navigationItemRegistry.ts

```diff
@@ -28,18 +28,19 @@ export type RootNavigationItem = NavigationItem & {
 };
 
 export class NavigationItemRegistry {
-    #items: RootNavigationItem[];
+    readonly #menus: Map<string, RootNavigationItem[]> = new Map();
 
-    constructor() {
-        this.#items = [];
-    }
-
-    add(navigationItems: RootNavigationItem[]) {
+    add(menuId: string, navigationItems: RootNavigationItem[]) {
```

Comment:
Minor detail here, but maybe we'll want to perform some minimal validation, like not accepting empty menus

Reply:
Could you clarify what would be the impact of accepting an empty menu?

Reply:
It's probably a developer mistake, so at least a warning would be nice. Anyway, not major

Reply:
Indeed, I usually try to not be too defensive for things without consequences. I am not sure what is the general consensus here for popular libraries.

I am also not sure what is the Workleap stance on this?

Reply:
I don't think we have a Workleap-wide stance on this, but like you said it's pretty inconsequential, so we can leave it as-is for now.